lib/withAuth.ts

```diff
@@ -33,12 +35,13 @@ export default function withAuth(handler: (req: NextApiRequestWithAuth, res: Nex
 
       // check if user exists
       await dbConnect();
-      const user = await UserModel.findOne<User>({ _id: req.userId });
+      const user = await UserModel.findOne<User>({ _id: req.userId }, '_id email isOfficial name score ts password', { lean: true });
```

Comment:
What is the purpose of this? Might as well just return the full object

Reply:
The full object returns a ton of random Mongo based keys (like `_v`, `__`, `$isNew` `_doc` `$__`) so `lean` makes it lean.